common/kernel/timing_log.cc

```diff
@@ -168,7 +165,27 @@ static void log_crit_paths(const Context *ctx, TimingResult &result)
         log_info("Critical path report for cross-domain path '%s' -> '%s':\n", start.c_str(), end.c_str());
         print_path_report(report);
     }
-};
+
+    // Min delay violated paths
+    // Show maximum of 10
+    auto num_min_violations = result.min_delay_violations.size();
+    if (num_min_violations > 0) {
+        log_break();
+        log_info("%zu Hold/min time violations (showing 10 worst paths):\n", num_min_violations);
+        for (size_t i = 0; i < std::min((size_t)10, num_min_violations); ++i) {
+            auto &report = result.min_delay_violations.at(i);
+            log_break();
+            std::string start = clock_event_name(ctx, report.clock_pair.start);
+            std::string end = clock_event_name(ctx, report.clock_pair.end);
+            if (report.clock_pair.start == report.clock_pair.end) {
+                log_nonfatal_error("Hold/min time violation for clock '%s':\n", start.c_str());
```

Comment:
we should follow https://github.com/YosysHQ/nextpnr/blob/4b63b1115e01590e1ca5ac7a7f9c17811455387e/common/kernel/timing_log.cc#L197-L199 and make this a warning too if `timing/allowFail` is set